cloud/openstack/os_router.py

```diff
@@ -235,6 +251,11 @@ def _build_kwargs(cloud, module, router, network):
         # can't send enable_snat unless we have a network
         kwargs['enable_snat'] = module.params['enable_snat']
 
+    if project_id:
+        kwargs['project_id'] = project_id
+    else:
+        kwargs['project_id'] = module.params['project']
```

Comment:
This is going to fail because you pop() project off in main(). I'm not even sure why you bother with this entire 'else' portion.